client/scripts/controllers/chain/substrate/identities.ts

```diff
@@ -100,20 +102,51 @@ class SubstrateIdentities implements StorageModule {
     // }
 
     // check on chain for registration we haven't seen yet & wait for it to appear
-    // if we have a super address to fetch, grab that instead
-    return this._Chain.query(
-      (api: ApiRx) => api.query.identity.identityOf<Option<Registration>>(acctToFetch.address)
-    ).pipe(
-      takeWhile((id) => !id.isSome, true),
-      map((id) => id.isSome
-        ? new SubstrateIdentity(this._Chain, this._Accounts, this, acctToFetch, id.unwrap())
-        : null),
+    let id = this._identityQueriesInProgress[who.address];
+    let identityExists$: Observable<boolean>;
+    if (!id) {
+      // Create a new "empty" identity and wait to see if it exists
+      id = new SubstrateIdentity(this._Chain, this._Accounts, this, who);
+      id.subscribe();
+      this._identityQueriesInProgress[who.address] = id;
+      identityExists$ = id.exists$;
+    } else {
+      // we've already created the identity and are waiting to see if it exists
+      identityExists$ = id.exists$;
+    }
+    return identityExists$.pipe(
+      filter((v) => typeof v === 'boolean'),
+      takeWhile((v) => !v, true),
```

Comment:
What is this `true` value doing, is that what is returned if !v?

Reply:
`true` here means if we see `v` then we return one final time (inclusive) and then close the observable, otherwise we keep returning `!v` whenever we see a new value. but we might be able to remove this entirely, based on how `identityExists$` returns. I'll look into it.

Reply:
Should I hold off on merging then?

Reply:
i'll look into this tomorrow morning and give you the OK

Reply:
Reviewed it again -- it should be fine as it is. We don't want to accidentally trigger the deletion twice.